Kentor.AuthServices/Configuration/CertificateElement.cs

```diff
@@ -109,39 +89,39 @@ public X509Certificate2 LoadCertificate()
             {
                 string fileName = FileName;
                 fileName = PathHelper.MapPath(fileName);
-                
+
                 return new X509Certificate2(fileName, "", X509KeyStorageFlags.MachineKeySet);
             }
             else
             {
                 // A 0 store location indicates that attributes to load from store are not present
                 // in the config.
-                if (StoreLocation != 0)
+                if (StoreLocation == 0)
+                {
+                    return null;
+                }
+
+                var store = new X509Store(StoreName, StoreLocation);
+                try
                 {
-                    var store = new X509Store(StoreName, StoreLocation);
                     store.Open(OpenFlags.ReadOnly);
-                    try
-                    {
-                        var certs = store.Certificates.Find(X509FindType, FindValue, false);
 
-                        if (certs.Count != 1)
-                        {
-                            throw new InvalidOperationException(
-                                string.Format(CultureInfo.InvariantCulture,
-                                "Finding cert through {0} in {1}:{2} with value {3} matched {4} certificates. A unique match is required.",
-                                X509FindType, StoreLocation, StoreName, FindValue, certs.Count));
-                        }

```

Comment:
Is this logic really changed? How is this related to your feature? I suggest reverting the changes to this entire file.